### ISSUE TITLE
Aggregate does not seem to work

### ISSUE THREAD

**Issue by tsievert** [johnschwenck/bp#9]
Hey,
I was testing the packages and noticed that the aggregate feature does not seem to work.
See reprex below, with data points well below the 3 min threshold.
The output seems to be the same, no matter if I include 'agg = TRUE' or not.

Please let me know if I missed anything obvious.

Thanks!

Edit: Improved reprex to have side by side example with agg = F and agg = T

    
``` r
library(bp)

data_raw_reprex<-tibble::tribble(
  ~DATE.TIME, ~SYST, ~DIAST,  ~HR,
  "2022-06-28 17:01:56", "129",   "76", "86",
  "2022-06-28 17:02:46", "138",   "76", "83",
  "2022-06-28 17:03:40", "130",   "66", "80",
  "2022-06-29 15:11:07", "127",   "77", "75",
  "2022-06-29 15:12:00", "124",   "79", "76",
  "2022-06-29 15:12:50", "127",   "76", "76",
  "2022-06-30 14:03:22", "134",   "81", "74",
  "2022-06-30 14:04:18", "139",   "81", "72",
  "2022-06-30 14:05:17", "130",   "77", "74",
  "2022-06-30 18:59:43", "734",   "74", "72",
  "2022-06-30 19:00:33", "131",   "78", "73",
  "2022-06-30 19:01:31", "146",   "72", "74",
  "2022-06-30 19:02:28", "125",   "73", "72"
)

data_proc_agg_F<-process_data(data_raw_reprex,
                              sbp = 'syst',
                              dbp = 'diast',
                              date_time = 'date.time',
                              hr = 'hr')
#> 1 values that exceeded the SUL or SLL thresholds were coerced to NA.
#> DATE column created from DATE_TIME column.
#> No PP column found or specified. Automatically generated from SBP and DBP columns.
#> No RPP column found or specified. Automatically generated from SBP and HR columns.
#> No MAP column found or specified. Automatically generated from SBP and DBP columns.

bp_report(data_proc_agg_F, save_report = 0)
#> Report not saved to a file. To save, specify save_report = TRUE.
```

![](https://i.imgur.com/VgHEPcK.png)

``` r
data_proc_agg_T<-process_data(data_raw_reprex,
                        sbp = 'syst',
                        dbp = 'diast',
                        date_time = 'date.time',
                        hr = 'hr',
                        agg = TRUE,
                        agg_thresh = '5',
                        collapse_df = TRUE)
#> 1 values that exceeded the SUL or SLL thresholds were coerced to NA.
#> DATE column created from DATE_TIME column.
#> No PP column found or specified. Automatically generated from SBP and DBP columns.
#> No RPP column found or specified. Automatically generated from SBP and HR columns.
#> No MAP column found or specified. Automatically generated from SBP and DBP columns.

bp_report(data_proc_agg_T, save_report = 0)
#> Report not saved to a file. To save, specify save_report = TRUE.
```

![](https://i.imgur.com/zIUuawt.png)

<sup>Created on 2022-07-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by johnschwenck** [comment]
Thank you for bringing this to our attention. After looking into the aggregation function, we realized the time differences between observations were not properly coerced to minutes via lubridate / difftime (in your reprex example, the differences were calculated as seconds, rendering the agg_thresh = 5 argument useless). We have made the necessary changes on our development branch and will push changes to GitHub (and CRAN) shortly. Will close this ticket when complete. We appreciate your patience.
